exercise_3/src/Teacher.cpp

```diff
@@ -0,0 +1,22 @@
+#include <sstream>
+#include <string>
+#include "Teacher.h"
+
+Teacher::Teacher(int &id, std::string &name) {
```

Comment:
1. "this->" is redundand.
2. You may initialize members on initialization list.
3. Make arguments const.